ARTICLE.md

```diff
@@ -1,27 +1,290 @@
-**Hayden Fuss**
+## Easier End-to-End TLS with OpenShift
+**Hayden Fuss**  
 **Rough Draft**
 
-### End-to-End Encryption in Kubernetes/OpenShift
+### Introduction
 
-Explain the difficulties, certain ways of solving it.
+One of the greatest features of OpenShift is its abstractions
+for exposing, securing, and properly load-balancing your web
+services.
 
-Describe the assumptions about the OpenShift configuration, and
-explain the autogenerated certificates feature.
+This series is meant for platform engineers and application
+developers who are new to OpenShift or looking to expand their
+knowledge. The goal is to cover some useful features you
+could leverage to implement end-to-end TLS for your web services
+as opposed to the more commonly used edge terminated TLS.
 
-### Overview of HTTPS with Spring Boot
+We will then dive a little deeper to discuss some practices you
+could use to automate this setup for your development teams, and
+in particular we will cover what is involved in doing so for
+Spring Boot applications. Lastly, we will go over how to handle
+certificate expiration and the difficulties that arise with
+Java keystores when doing so.
+
+There will be links and code snippets throughout the articles,
+but if you want something you could get your hands on check out
+the [sample code](https://github.com/hfuss/e2e-encryption-openshift). For
+these examples, we will assume our OpenShift cluster
+administrators have taken care of two important details:
+
+1. There is a `*.apps.example.com` DNS record pointed
+to the cluster's load-balancer.
+2. The cluster's load-balancer has a valid TLS certificate for `*.apps.example.com`.
+
+If this is not the case for your cluster, work with your cluster administrators in getting a similar setup made and have them consider setting up tools such as [External DNS]() and [Certificate Manager]().
+
+### Part One: Re-encrypt TLS and Automated Internal Certificates with OpenShift
+
+#### Introducing the `Route`
+
+So first things first, what are these great OpenShift
+abstractions for exposing and load-balancing web services and
+how do they differ from what else exists in the Kubernetes
+landscape?
+
+Before there was the `Ingress` in Kubernetes, the OpenShift
+developers had provided `Route`'s. Both make managing external
+access to HTTP/HTTPS `Service`'s running on a cluster
+relatively easy. However, there are some notable differences,
+see the [Appendix](#ingress-vs-route) for a more in-depth
+comparison between `Route`'s and `Ingress`'s if you are
+interested.
+
+Overall, a `Route` in some ways is a slightly simpler
+abstraction which provides more robust support for handling
+TLS, and a greater separation of duties between the cluster
+administrators and developers. Additionally, with OpenShift
+3.10 there is now an ingress controller which will
+automatically make a `Route` (or possibly multiple) in the
+background for any `Ingress`'s you define.
```

Comment:
I finished this section and didn't come away with any idea of what `Ingress` is. Only one sentence introduces the route in this section, and that's by comparing it to `Ingress`. If you expect your reader to have k8s knowledge but not Openshift, you should highlight that in the opening paragraphs